CRM/Core/BAO/File.php

```diff
@@ -261,51 +261,32 @@ public static function deleteEntityFile($entityTable, $entityID, $fileTypeID = N
     [$sql, $params] = self::sql($entityTable, $entityID, $fileTypeID, $fileID);
     $dao = CRM_Core_DAO::executeQuery($sql, $params);
 
-    $cfIDs = [];
-    $cefIDs = [];
+    $cfIDs = $cefIDs = [];
     while ($dao->fetch()) {
       $cfIDs[$dao->cfID] = $dao->uri;
-      $cefIDs[] = $dao->cefID;
+      $cefIDs[$dao->cfID] = $dao->cefID;
```

Comment:
1. Compose an email in civi to two recipients.
2. Add an attachment.
3. Send the email.
4. Go find one of the activities and try to delete that activity.
5. It gives an error: `DELETE FROM civicrm_file WHERE ( civicrm_file.id = 2 ) [nativecode=1451 ** Cannot delete or update a parent row: a foreign key constraint fails (civicrm_entity_file, CONSTRAINT FK_civicrm_entity_file_file_id FOREIGN KEY (file_id) REFERENCES civicrm_file (id))]`

Reply:
Thanks @demeritcowboy for your feedback, looking at it.

Reply:
@monishdeb any updates?

Reply:
I can't see a way to do this in the UI, but since $cefIDs is now keyed on the file id this, if the loop iterates more than once it could theoretically overwrite the value from a previous loop iteration. But you'd have to do something like call this with just entitytable set and no entity id, or if entitytable is civicrm_file itself. I'm not sure if that happens anywhere. 🤷‍♂️